src/renderer/state/upload/logics.ts

```diff
@@ -724,25 +730,29 @@ const uploadWithoutMetadataLogic = createLogic({
         deps.action.payload
       );
       uploads = await Promise.all(
-        filePaths.map((filePath) =>
-          deps.fms.initiateUpload(
-            {
-              file: {
-                disposition: "tape", // prevent czi -> ome.tiff conversions
-                fileType:
-                  extensionToFileTypeMap[
-                    path.extname(filePath).toLowerCase()
-                  ] || FileType.OTHER,
-                originalPath: filePath,
-                shouldBeInArchive: true,
-                shouldBeInLocal: true,
-              },
-              microscopy: {},
-            },
-            user,
-            { groupId }
-          )
-        )
+          filePaths.flat().map((filePath) => {
+            const isMultifile = determineIsMultifile(filePath); // todo: simplify this whole function so this isn't called twice
+            return deps.fms.initiateUpload(
+                {
+                  file: {
+                    disposition: "tape", // prevent czi -> ome.tiff conversions
+                    fileType:
+                        extensionToFileTypeMap[
+                            path.extname(filePath).toLowerCase()
+                            ] || FileType.OTHER,
+                    originalPath: filePath,
+                    shouldBeInArchive: true,
+                    shouldBeInLocal: true,
+                  },
+                  microscopy: {},
+                },
+                user,
+                {
+                  groupId,
+                  multifile: isMultifile
```

Comment:
Does this undergo another transformation to create the request body?  Asking because I expected the  `multifile` arg to be at the same level as `shouldBeInArchive` and `shouldBeInLocal`

Reply:
The request body is constructed in the file-storage-service module. I honestly wasn't sure how this object was supposed to be structured and was vaguely aware that the `service_fields` were a grab-bag... Should I move it up to `file`?